File: solutions/python/bob/1/bob.py
def response(hey_bob):
    str = hey_bob.strip()

    if not str:
        return "Fine. Be that way!"

    isquestion = str.endswith("?")
    isshout = any(c.isalpha() for c in str) and str == str.upper()

    res = ""
    match (isquestion, isshout):
        case (True, True):
            res = "Calm down, I know what I'm doing!" 
        case (_, True):
            res = "Whoa, chill out!"
        case (True, _):
            res = "Sure."
        case _:
            res = "Whatever."
    return res

File: solutions/kotlin/bob/1/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        val say: String = input.trim()
        if (say.isBlank()) return "Fine. Be that way!"

        val isQuestion: Boolean = say.endsWith("?")
        val isYell: Boolean = say.uppercase() == say && say.any { c: Char -> c in 'A'..'Z' }

        return when (Pair(isQuestion, isYell)) {
            Pair(true, true) -> "Calm down, I know what I'm doing!"
            Pair(true, false) -> "Sure."
            Pair(false, true) -> "Whoa, chill out!"
            else -> "Whatever."
        }
    }
}
